### DISCUSSION TITLE
Errori in installazione HA su Win10

### DISCUSSION THREAD

**Discussion by Stefano2023** [indomus/forum#2247] [HUB: Home Assistant]
E' il mio primo approccio a questo mondo e chiedo scusa anticipatamente se farò perdere del tempo a qualcuno che vorrà aiutarmi. Grazie fin d'ora.
Dopo aver installato quanto indicato nelle guide di indomus per il mondo Windows, al lancio del comando di installazione, mi ritrovo con  gli errori indicati sotto:

C:\Users\stefa>pip3 install homeassistant

Collecting homeassistant
  Using cached homeassistant-2023.1.1-py3-none-any.whl (21.1 MB)
Collecting aiohttp==3.8.1
  Using cached aiohttp-3.8.1.tar.gz (7.3 MB)
.......
Collecting anyio<5.0,>=3.0
  Using cached anyio-3.6.2-py3-none-any.whl (80 kB)
Building wheels for collected packages: aiohttp, bcrypt, lru-dict, yarl
**Building wheel for aiohttp (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for aiohttp (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [106 lines of output]
      *********************
      * Accelerated build *
      *********************
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
      creating build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\abc.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\base_protocol.py -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\client.py -> build\lib.win-amd64-cpython-311\aiohttp
........
      copying aiohttp\__init__.py -> build\lib.win-amd64-cpython-311\aiohttp
      running egg_info
      writing aiohttp.egg-info\PKG-INFO
      writing dependency_links to aiohttp.egg-info\dependency_links.txt
      writing requirements to aiohttp.egg-info\requires.txt
      writing top-level names to aiohttp.egg-info\top_level.txt
      reading manifest file 'aiohttp.egg-info\SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching 'aiohttp' anywhere in distribution
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '*.pyd' found anywhere in distribution
      warning: no previously-included files matching '*.so' found anywhere in distribution
      warning: no previously-included files matching '*.lib' found anywhere in distribution
      warning: no previously-included files matching '*.dll' found anywhere in distribution
      warning: no previously-included files matching '*.a' found anywhere in distribution
      warning: no previously-included files matching '*.obj' found anywhere in distribution
      warning: no previously-included files found matching 'aiohttp\*.html'
      no previously-included directories found matching 'docs\_build'
      adding license file 'LICENSE.txt'
      writing manifest file 'aiohttp.egg-info\SOURCES.txt'
      copying aiohttp\_cparser.pxd -> build\lib.win-amd64-cpython-311\aiohttp
      copying aiohttp\_find_header.c -> build\lib.win-amd64-cpython-311\aiohttp
..........
      creating build\temp.win-amd64-cpython-311\Release
      creating build\temp.win-amd64-cpython-311\Release\aiohttp
      "C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\Users\stefa\AppData\Local\Programs\Python\Python311\include -IC:\Users\stefa\AppData\Local\Programs\Python\Python311\Include "-IC:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30133\include" /Tcaiohttp/_websocket.c /Fobuild\temp.win-amd64-cpython-311\Release\aiohttp/_websocket.obj_websocket.c
      C:\Users\stefa\AppData\Local\Programs\Python\Python311\include\pyconfig.h(59): fatal error C1083: Non Š possibile aprire il file inclusione: 'io.h': No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for aiohttp
  Building wheel for bcrypt (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for bcrypt (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [24 lines of output]
      C:\Users\stefa\AppData\Local\Temp\pip-build-env-jid574rx\overlay\Lib\site-packages\setuptools\config\setupcfg.py:508: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build\lib.win-amd64-cpython-311
.........
      C:\Users\stefa\AppData\Local\Programs\Python\Python311\include\pyconfig.h(59): fatal error C1083: Non Š possibile aprire il file inclusione: 'io.h': No such file or directory
      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for yarl
Failed to build aiohttp bcrypt lru-dict yarl
ERROR: Could not build wheels for aiohttp, bcrypt, yarl, which is required to install pyproject.toml-based projects

C:\Users/stefa>
C:\Users/stefa>

Qualcuno potrebbe gentilmente darmi una dritta e/o indirizzarmi a qualche link che suggerisca soluzioni? Personalmente, nel forum non ho trovato indicazioni utili.

Grazie e buona serata

**Comment by snakuzzo** [comment]
L'unica dritta che ti posso dare è...lascia perdere questo tipo di installazione se non sai un minimo smanettare.
Se vuoi provare su Windows puoi farlo con i tipi di installazione "Core su Docker" oppure tiri su una VM con HA OS
